### DISCUSSION TITLE
Documentation Generation

### DISCUSSION THREAD

**Discussion by ydaveluy** [ThalesGroup/xsmp-modeler-core#21] [Ideas]
How to integrate in XSMP documentation generation ?
- Multiplatform : Linux/Windows/MacOS ?
- Based on document templates ?
- Techno: Word/OpenOffice/Latex/… ?
- List of use cases
- Design process / work sharing within the community ?

Any idea or feedback is welcome !

**Comment by aingenito-telespazio** [comment]
I can open with a few question:
* how easy/complicated would be to plug your how templates in the tool?
* are (uml-like) diagrams an option?

Then I can say that we are quite happy with static-site generation. We are current using mkdocs and the python-markdown dialect, but there are many options.

Word has always been cumbersome, at least to integrate hand-written and generated parts.

**Comment by ydaveluy** [comment]
Thanks for your feedback !

[UML Designer](https://www.umldesigner.org/ref-doc/define-the-system.html) is based on Sirius. It could be possible to reuse its diagrams in XSMP. It seems possible to export [images](https://www.eclipse.org/forums/index.php/t/1103425/) from sirius.

**Reply by TristanFAURE** [reply]
take care UML designer based on Sirius is quite huge. Maybe some plantUML like technologies could be easier to maintain ?
(edit : i see you already have sirius with ecore tools, so why not UML designer but it is no longer maintained (no more activities on the github)
